homeassistant/components/unifiprotect/migrate.py

```diff
@@ -30,6 +35,32 @@ async def async_migrate_data(
     await async_migrate_device_ids(hass, entry, protect)
     _LOGGER.debug("Completed Migrate: async_migrate_device_ids")
 
+    entity_registry = er.async_get(hass)
+    created_issue = False
+    for entity in er.async_entries_for_config_entry(entity_registry, entry.entry_id):
+        if (
+            entity.domain == Platform.SENSOR
+            and entity.disabled_by is None
+            and "detected_object" in entity.unique_id
+            and (
+                automations_with_entity(hass, entity.entity_id)
+                or scripts_with_entity(hass, entity.entity_id)
+            )
+        ):
+            created_issue = True
+            ir.async_create_issue(
+                hass,
+                DOMAIN,
+                "deprecate_smart_sensor",
+                is_fixable=False,
+                breaks_in_ha_version="2023.2.0",
+                severity=IssueSeverity.WARNING,
+                translation_key="deprecate_smart_sensor",
+            )
+            break
+    if not created_issue:
+        ir.async_delete_issue(hass, DOMAIN, "deprecate_smart_sensor")
```

Comment:
Should this listen to automation or script update events, so it can delete it when the issue is resolved?

Reply:
Added for automation reloads, but it does not seem like the script domain fires the same event that the automation does.

Reply:
Thanks @AngellusMortis